ui/src/app/views/workspace-list/component.ts

```diff
@@ -41,27 +46,34 @@ export class WorkspaceListComponent implements OnInit, OnDestroy {
       }
       if (profile.freeTierBillingProjectStatus === BillingProjectStatus.Ready) {
         this.billingProjectInitialized = true;
+        // Only once we know the billing project status do we request/display
+        // workspaces for two reasons:
+        // - If the FC user is not yet initialized, getWorkspaces() may fail
+        //   with a 401.
+        // - While the billing project is being initialized, we want to keep the
+        //   big spinner on the page to provide obvious messaging to the user
+        //   about the expected wait time.
+        this.workspacesService.getWorkspaces()
```

Comment:
I think we may want to load on pending as well, because a user can be shared on workspaces that they don't own, and start work on those.

That said, that would mean less info to the user at the cost of that so... maybe not?

Tagging in @fknabe

Reply:
We actually can't let them continue to the workspaces page currently, things will fail if they don't have a billing project at that point. Also I think longer term this is probably OK - once we have the initial sign up flow fully working, we will have already been initializing their project as they go through tos, training, ID verification, etc.

Reply:
Got it! For some reason I thought it only failed if they weren't initialized at all in FC, regardless of Billing Project status. In that case, 👍 from me.

Reply:
Hmm, so `getWorkspace()` will probably work with just a user, but `createCluster()` probably won't. I haven't tried that failure mode yet, but I'm not sure I want to find out what happens.